app/models/journeyDomain/transport/AuthorisationDomain.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.journeyDomain.transport
+
+import cats.implicits.catsSyntaxTuple2Semigroupal
+import models.ProcedureType._
+import models.domain.{GettableAsReaderOps, UserAnswersReader}
+import models.journeyDomain.Stage.{AccessingJourney, CompletingJourney}
+import models.journeyDomain.{JourneyDomainModel, Stage}
+import models.transport.authorisations.AuthorisationType
+import models.transport.authorisations.AuthorisationType.{ACR, TRD}
+import models.transport.transportMeans.departure.InlandMode._
+import models.{Index, Mode, UserAnswers}
+import pages.preTaskList.ProcedureTypePage
+import pages.traderDetails.consignment.ApprovedOperatorPage
+import pages.transport.authorisation.index.{AuthorisationReferenceNumberPage, AuthorisationTypePage}
+import pages.transport.transportMeans.departure.InlandModePage
+import play.api.mvc.Call
+
+case class AuthorisationDomain(authorisationType: AuthorisationType, referenceNumber: String)(index: Index) extends JourneyDomainModel {
+
+  def asString: String =
+    AuthorisationDomain.asString(authorisationType, referenceNumber)
+
+  override def routeIfCompleted(userAnswers: UserAnswers, mode: Mode, stage: Stage): Option[Call] = Some {
+    stage match {
+      case AccessingJourney =>
+        if (userAnswers.get(AuthorisationTypePage(index)).isEmpty && index.isFirst) {
+          controllers.transport.authorisations.index.routes.AuthorisationReferenceNumberController.onPageLoad(userAnswers.lrn, mode, index)
+        } else {
+          controllers.transport.authorisations.index.routes.AuthorisationTypeController.onPageLoad(userAnswers.lrn, mode, index)
+        }
+      case CompletingJourney =>
+        controllers.transport.authorisations.routes.AddAnotherAuthorisationController.onPageLoad(userAnswers.lrn, mode)
+    }
+  }
+}
+
+object AuthorisationDomain {
+
+  def asString(authorisationType: AuthorisationType, referenceNumber: String): String =
+    s"${authorisationType.toString} - $referenceNumber"
```

Comment:
Again this is minor but calling `.toString` on an authorisation type takes the `WithName` parameter which might not always be what we're after. We have other examples like https://github.com/hmrc/manage-transit-movements-departure-frontend/blob/15c85fd6d3dd0c6406c8851348d3bffbeea023eb/app/models/transport/supplyChainActors/SupplyChainActorType.scala#L24-L25